### PR TITLE
Sequelize

### LINKED ISSUE #6: Database Setup: Complete migration/model constraints/references
Set up server to add new user information from the client side form to the database

### LINKED ISSUE #7: Database Setup: Create Models
Set up a registration form to send information to the server to create a new user account

### PR DESCRIPTION
**Pull request by WJVincent**
Merge sequelize files into main.

### CODE REVIEW

File: db/migrations/20201130231152-create-user.js [hunk 1]
```diff
@@ -8,18 +8,19 @@ module.exports = {
...
       email: {
         allowNull: false,
-        type: Sequelize.STRING(75),
+        type: Sequelize.STRING(250),
         unique:true
       },
       password: {
         allowNull: false,
-        type: Sequelize.STRING(50),
+        type: Sequelize.BYTEA(250),
```
Comment (ghost) [inline db/migrations/20201130231152-create-user.js]: Curious about this decision. Can you tell me more about why BYTEA and not STRING?

File: db/migrations/20201130231154-create-user-book.js [hunk 2]
```diff
@@ -26,7 +26,8 @@ module.exports = {
       },
       status: {
         allowNull: false,
-        type: Sequelize.BOOLEAN
+        type: Sequelize.STRING(10),
+        unique: true
```
Comment (ghost) [inline db/migrations/20201130231154-create-user-book.js]: `unique: true` is going to make it so you cannot have the same status in **any other row in this user book table**. I don't _think_ this is what you want to do, but help me understand if I'm wrong!

File: db/models/author.js [hunk 3]
```diff
@@ -4,7 +4,8 @@ module.exports = (sequelize, DataTypes) => {
     name: DataTypes.STRING
   }, {});
   Author.associate = function(models) {
-    // associations can be defined here
-  };
-  return Author;
-};
\ No newline at end of file
+    Author.belongsToMany(models.Book,{through: models.BookAuthor})
```
Comment (ghost) [inline db/models/author.js:7]: Do you not have to provide `foreignKey` and `otherKey` on the configuration object (second argument of `belongsToMany`)?

File: db/models/user.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-    name: DataTypes.STRING,
+    userName: DataTypes.STRING,
     email: DataTypes.STRING,
-    password: DataTypes.STRING
+    password: DataTypes.BYTEA
   }, {});
   User.associate = function(models) {
-    // associations can be defined here
+    User.belongsToMany(models.Book, {through:models.UserBook})
+    User.belongsToMany(models.User, {as:'friendId', through:models.Friend})
```
Comment (ghost) [inline db/models/user.js]: I think your `as` here should not be `"friendId"` but maybe like `"friends"`?

**Review by ghost — DISMISSED** [review]
This is looking pretty good, but I left some comments to be addressed. 😄

File: db/migrations/20201130231152-create-user.js (hunk 1 above)
Comment (WJVincent) [reply db/migrations/20201130231152-create-user.js]: Yeah, just realized we actually changed this again, so we will need to alter this pull request regardless. It is currently listed as STRING.BINARY. I'm fairly unclear on the actual datatype that needs to be stored in this location as the hashed password value. Would definitely appreciate some clarification here.

File: db/migrations/20201130231154-create-user-book.js (hunk 2 above)
Comment (WJVincent) [reply db/migrations/20201130231154-create-user-book.js]: Yeah, this is an error, good catch!

File: db/models/author.js (hunk 3 above)
Comment (WJVincent) [reply db/models/author.js:7]: I believe that this syntax recognizes the foriegn key and the other key based on the originating model and the referenced model, by including the joins model as the keyed value of through. Could definitely be wrong. I believe this was changed by our reading through the documentation with you and Chris yesterday.

File: db/models/user.js (hunk 4 above)
Comment (WJVincent) [reply db/models/user.js]: Yeah, I was unclear on this one, we have this value on the table as friendId. Will that automatically recognize this as reference or should we change that reference on the table?

File: db/migrations/20201130231152-create-user.js (hunk 1 above)
Comment (ghost) [reply db/migrations/20201130231152-create-user.js]: Hashed passwords are just strings, so you can just use Sequelize.STRING here. Your `bcrypt` (or possibly whatever algorithm you are using) hashes will always be the same length, so you can make the character limit here based on the length of those hash strings.

File: db/models/author.js (hunk 3 above)
Comment (ghost) [reply db/models/author.js:7]: Let me know if the `foreignKey`/`otherKey` is implied here and if it works, very curious!

File: db/models/user.js (hunk 4 above)
Comment (ghost) [reply db/models/user.js]: WITHOUT the usage of aliasing (i.e. `as`), this relationship will allow you to do:

```js
const user = User.findbyPk(1, { include: { model: User } });
console.log(user.users);
```

WITH aliasing set-up on your model, you can do something like this:

```js
// "as" key here HAS TO BE the same as what you've configure on your model `associate`
const user = User.findbyPk(1, { include: { model: User, as: "friends" } }); 
console.log(user.friends);
```

File: db/models/author.js (hunk 3 above)
Comment (WJVincent) [reply db/models/author.js:7]: Okay sounds good, I'll let you know if it works or if it breaks :)

**Review by RamsesRomeroJr — APPROVED** [review]
Looks good, good job team keep up the work

**Pull request merged**